### PR TITLE
wasm client: rpc stream updates

### PR DESCRIPTION
**Pull request by maan2003**
makes stream cancellable
sends end event for stream

based on requests from frontend. 

see commits

### CODE REVIEW

**Comment by Kodylow** [comment]
@alexlwn123

**Review by alexlwn123 — APPROVED** [review]
Seems to work great in the web sdk

File: fedimint-client-wasm/src/lib.rs [hunk 1]
```diff
@@ -93,30 +107,60 @@ impl WasmClient {
...
+            let future = async {
+                let mut stream = pin!(Self::rpc_inner(&client, &module, &method, payload));
+
+                while let Some(item) = stream.next().await {
+                    let this = JsValue::null();
+                    let _ = match item {
+                        Ok(item) => cb.call1(
+                            &this,
+                            &JsValue::from_str(
+                                &serde_json::to_string(&json!({"data": item})).unwrap(),
```
Comment (bradleystachurski) [inline fedimint-client-wasm/src/lib.rs:137]: Will changing the json key to `"data"` cause issue? We were previously sending `&serde_json::to_string(&item)`

**Review by bradleystachurski — APPROVED** [review]

File: fedimint-client-wasm/src/lib.rs (hunk 1 above)
Comment (maan2003) [reply fedimint-client-wasm/src/lib.rs:137]: yes, this was requested by @alexlwn123

File: fedimint-client-wasm/src/lib.rs [hunk 2]
```diff
@@ -93,30 +107,60 @@ impl WasmClient {
...
+                let _ = cb.call1(
+                    &JsValue::null(),
+                    &JsValue::from_str(&serde_json::to_string(&json!({"end": null})).unwrap()),
+                );
             };
-        }
-    }
 
+            let abortable_future = Abortable::new(future, abort_registration);
+            let _ = abortable_future.await;
```
Comment (elsirion) [inline fedimint-client-wasm/src/lib.rs:157]: TIL :exploding_head:

**Review by elsirion — COMMENTED** [review]
:tada:

**Pull request merged**